pkg/ensurance/analyzer/analyzer.go

```diff
@@ -187,332 +187,161 @@ func (s *AnormalyAnalyzer) Analyze(state map[string][]common.TimeSeries) {
+		(common.GetValueByName(ts.Labels, common.LabelNamePodNamespace) != "")
+}
 
-		for _, ts := range series {
-			triggered := s.evaluator.EvalWithMetric(object.MetricRule.Name, float64(object.MetricRule.Value.Value()), ts.Samples[0].Value)
+func (s *AnormalyAnalyzer) getSeries(state []common.TimeSeries, selector *metav1.LabelSelector, metricName string) ([]common.TimeSeries, error) {
+	series := s.getTimeSeriesFromMap(state, selector)
+	if len(series) == 0 {
+		return []common.TimeSeries{}, fmt.Errorf("time series length is 0 for metric %s", metricName)
+	}
+	return series, nil
+}
 
-			klog.V(6).Infof("Anormaly detection result %v, Name: %s, Value: %.2f, %s/%s", triggered,
-				object.MetricRule.Name,
-				ts.Samples[0].Value,
-				common.GetValueByName(ts.Labels, common.LabelNamePodNamespace),
-				common.GetValueByName(ts.Labels, common.LabelNamePodName))
+func (s *AnormalyAnalyzer) trigger(series []common.TimeSeries, object ensuranceapi.ObjectiveEnsurance) (bool, []types.NamespacedName) {
+	var impacted []types.NamespacedName
+	var triggered, threshold bool
+	for _, ts := range series {
+		triggered = s.evaluator.EvalWithMetric(object.MetricRule.Name, float64(object.MetricRule.Value.Value()), ts.Samples[0].Value)
 
-			if !triggered {
-				continue
-			}
+		klog.V(6).Infof("Anormaly detection result %v, Name: %s, Value: %.2f, %s/%s", triggered,
+			object.MetricRule.Name,
+			ts.Samples[0].Value,
+			common.GetValueByName(ts.Labels, common.LabelNamePodNamespace),
+			common.GetValueByName(ts.Labels, common.LabelNamePodName))
 
-			if (common.GetValueByName(ts.Labels, common.LabelNamePodName) != "") &&
-				(common.GetValueByName(ts.Labels, common.LabelNamePodNamespace) != "") {
-				pod, err := s.podLister.Pods(common.GetValueByName(ts.Labels, common.LabelNamePodNamespace)).Get(common.GetValueByName(ts.Labels, common.LabelNamePodName))
-				if err != nil {
-					klog.V(4).Infof("Warning: analyze: Pod %s/%s not found", common.GetValueByName(ts.Labels, common.LabelNamePodNamespace), common.GetValueByName(ts.Labels, common.LabelNamePodName))
-					continue
-				} else {
-					var qosPriority = executor.ClassAndPriority{PodQOSClass: pod.Status.QOSClass, PriorityClassValue: utils.GetInt32withDefault(pod.Spec.Priority, 0)}
-					if qosPriority.Greater(basicThrottleQosPriority) {
-						impacted = append(impacted, types.NamespacedName{Name: common.GetValueByName(ts.Labels, common.LabelNamePodName),
-							Namespace: common.GetValueByName(ts.Labels, common.LabelNamePodNamespace)})
-						threshold = true
-					}
+		if triggered {
+			if s.isContainerMetric(object, ts) {
+				influced := s.getImpacted(ts)
+				if len(influced) > 0 {
+					threshold = true
+					impacted = append(impacted, influced...)
 				}
 			} else {
-				// node metrics
-				klog.Infof("PodName or PodNamespace is empty")
 				threshold = true
+				return threshold, []types.NamespacedName{}
```

Comment:
not return here， return the end of the function ?

Reply:
Done